src/components/header/index.js

```diff
@@ -0,0 +1,178 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import OutsideClickHandler from 'react-outside-click-handler';
+
+import { Media, MediaContextProvider, mediaStyles } from 'utils/responsive';
+import { APP_URL } from 'utils/constants';
+
+import gfwLogo from 'assets/logos/gfw.png';
+import MenuIcon from 'assets/icons/menu.svg';
+import CloseIcon from 'assets/icons/close.svg';
+
+import NavLink from 'components/header/components/nav-link';
+import NavMenu from './components/nav-menu';
+import NavAlt from './components/nav-alt';
+import SubmenuPanel from './components/submenu-panel';
+
+import defaultConfig from './config';
+
+import './styles.scss';
+
+class Header extends PureComponent {
+  static propTypes = {
+    className: PropTypes.string,
+    loggedIn: PropTypes.bool,
+    loggingIn: PropTypes.bool,
+    setQueryToUrl: PropTypes.func,
+    NavLinkComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
+    openContactUsModal: PropTypes.func,
+    appUrl: PropTypes.string,
+    navMain: PropTypes.array,
+    relative: PropTypes.bool,
+    languages: PropTypes.array,
+    pathname: PropTypes.string,
+  };
+
+  static defaultProps = {
+    className: '',
+    loggedIn: false,
+    appUrl: APP_URL,
+    ...defaultConfig,
+  };
+
+  state = {
+    pathname: this.props.pathname || '',
+    showSubmenu: false,
+    clickOutside: false,
+    lang: 'en',
+    languages: this.props.languages,
+  };
+
+  componentDidMount() {
+    if (!this.props.pathname && typeof window !== 'undefined') {
+      this.setState({ pathname: window.location.pathname });
+      const $style = document.createElement('style');
+      document.head.appendChild($style);
+      $style.type = 'text/css';
+      $style.innerHTML = mediaStyles;
```

Comment:
what is the purpose of this? will we have to include it in every high-level component?

Reply:
Ah yes I forgot about this bit. So the mediaStyles helper outputs css of the breakpoints from the MediaContext provider. For SSR they recommend adding it to you document as fall back for SSR when you dont have access to screen size. I will try to think of a better solution for this as otherwise you have to manually add it in your app.

Reply:
One solution, which is horrible, is to copy and paste the output of to the global styles.